subsys/bluetooth/audio/pbp.c

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright 2023 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/kernel.h>
+#include <zephyr/sys/byteorder.h>
+#include <zephyr/types.h>
+#include <zephyr/device.h>
+#include <zephyr/init.h>
+#include <stdlib.h>
+#include <zephyr/bluetooth/bluetooth.h>
+#include <zephyr/bluetooth/conn.h>
+#include <zephyr/bluetooth/gatt.h>
+#include <zephyr/bluetooth/audio/audio.h>
+#include <zephyr/bluetooth/audio/pbp.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/sys/check.h>
+
+LOG_MODULE_REGISTER(bt_pbp, CONFIG_BT_PBP_LOG_LEVEL);
+
+/* PBA Service UUID + Public Broadcast Announcement features + Public_Info metadata of length 1 */
+#define MIN_PBA_SIZE			5
+
+/* No audio configuration set */
+#define PBA_FEATURES_INVALID_VALUE	0
+
+static enum bt_pbp_announcement_feature stream_config_preferences = PBA_FEATURES_INVALID_VALUE;
+
+static bool parse_cb(struct bt_data *data, void *user_data)
+{
+	bool *found = (bool *)(user_data);
+
+	if (data->type == BT_AUDIO_METADATA_TYPE_PROGRAM_INFO) {
+		*found = true;
+
+		return false;
+	}
+
+	return true;
+}
+
+void bt_pbp_init_source(enum bt_pbp_announcement_feature features)
+{
+	stream_config_preferences = features;
+}
```

Comment:
Hmm, the entire purpose of `bt_pbp_init_source` seems to be to just stored a local feature variable. 

Couldn't this variable just be added to `bt_pbp_create_public_broadcast_announcement` and then we can remove the local storage as well as `bt_pbp_init_source`?